test/built-ins/TypedArrays/from/custom-ctor-returns-other-instance.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 22.2.2.1
+esid: sec-%typedarray%.from
+description: >
+  Custom constructor can return any TypedArray instance with higher or same
+  length
+info: |
+  %TypedArray%.from ( source [ , mapfn [ , thisArg ] ] )
+
+  ...
+  7. If usingIterator is not undefined, then
+    a. Let values be ? IterableToList(source, usingIterator).
+    b. Let len be the number of elements in values.
+    c. Let targetObj be ? TypedArrayCreate(C, «len»).
+  ...
+  10. Let len be ? ToLength(? Get(arrayLike, "length")).
+  11. Let targetObj be ? TypedArrayCreate(C, « len »).
+  ...
+includes: [testTypedArray.js]
+features: [Symbol.iterator]
+---*/
+
+var sourceItor = [1, 2];
+var sourceObj = {
+  length: 2
+};
+
+testWithTypedArrayConstructors(function(TA) {
+  var result;
+  var Other = TA === Int8Array ? Uint8Array : Int8Array;
+  var custom = new Other(2);
+  var ctor = function() {
+    return custom;
+  };
+
+  result = TypedArray.from.call(ctor, sourceItor);
```

Comment:
Should this be `TA.from`? Otherwise this test isn't really making use of `TA` at all.